### PR TITLE
test: x/oracle keeper

### PR DESCRIPTION
**Pull request by adamewozniak**
## Description

Adds keeper_tests for the x/oracle module

----

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [x] added appropriate labels to the PR
- [x] added `!` to the type prefix if API or client breaking change
- [x] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: app/beta/app_beta.go [hunk 1]
```diff
@@ -479,6 +497,7 @@ func New(
 		ibchost.ModuleName,
 		leveragetypes.ModuleName,
 		peggytypes.ModuleName,
+		oracletypes.ModuleName,
```
Comment (alexanderbez) [inline app/beta/app_beta.go:500]: let's put this _after_ leverage.

File: app/beta/app_beta.go [hunk 2]
```diff
@@ -487,6 +506,7 @@ func New(
 		leveragetypes.ModuleName,
 		stakingtypes.ModuleName,
 		peggytypes.ModuleName,
+		oracletypes.ModuleName,
```
Comment (alexanderbez) [inline app/beta/app_beta.go:509]: ditto, put this _after_ leverage.

File: x/oracle/keeper/keeper_test.go [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+
+	s.app.OracleKeeper.SetFeederDelegation(ctx, valAddr, feederAddr)
+	s.Require().Equal(app.OracleKeeper.GetFeederDelegation(ctx, valAddr), feederAddr)
+}
+
+func (s *IntegrationTestSuite) Test_MissCounter() {
+	app, ctx := s.app, s.ctx
+	missCounter := uint64(rand.Intn(100))
+
+	s.Require().Equal(app.OracleKeeper.GetMissCounter(ctx, valAddr), uint64(0x0))
```
Comment (toteki) [inline x/oracle/keeper/keeper_test.go:134]:
```suggestion
	s.Require().Equal(app.OracleKeeper.GetMissCounter(ctx, valAddr), uint64(0))
```

File: x/oracle/keeper/keeper_test.go [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+	app.OracleKeeper.SetMissCounter(ctx, valAddr, missCounter)
+	s.Require().Equal(app.OracleKeeper.GetMissCounter(ctx, valAddr), missCounter)
+
+	app.OracleKeeper.DeleteMissCounter(ctx, valAddr)
+	s.Require().Equal(app.OracleKeeper.GetMissCounter(ctx, valAddr), uint64(0x0))
```
Comment (toteki) [inline x/oracle/keeper/keeper_test.go:139]:
```suggestion
	s.Require().Equal(app.OracleKeeper.GetMissCounter(ctx, valAddr), uint64(0))
```

File: x/oracle/keeper/keeper_test.go [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+	prevote := types.AggregateExchangeRatePrevote{
+		Hash:        "hash",
+		Voter:       addr.String(),
+		SubmitBlock: 0,
+	}
+	app.OracleKeeper.SetAggregateExchangeRatePrevote(
+		ctx,
+		valAddr,
+		prevote,
+	)
```
Comment (toteki) [inline x/oracle/keeper/keeper_test.go:154]:
```suggestion
	app.OracleKeeper.SetAggregateExchangeRatePrevote(ctx, valAddr, prevote)
```

File: x/oracle/keeper/keeper_test.go [hunk 6]
```diff
@@ -0,0 +1,255 @@
...
+
+	_, err := app.OracleKeeper.GetAggregateExchangeRatePrevote(
+		ctx,
+		valAddr,
+	)
```
Comment (toteki) [inline x/oracle/keeper/keeper_test.go:159]:
```suggestion
	_, err := app.OracleKeeper.GetAggregateExchangeRatePrevote(ctx, valAddr)
```

File: x/oracle/keeper/keeper_test.go (hunk 5 above)
Comment (alexanderbez) [reply x/oracle/keeper/keeper_test.go:154]: Good catch. @adamewozniak we don't need to do multi-line breaks unless it exceeding the linting setting.

File: x/oracle/keeper/keeper_test.go [hunk 7]
```diff
@@ -0,0 +1,255 @@
...
+	s.Require().NoError(err)
+
+	app.OracleKeeper.DeleteAggregateExchangeRatePrevote(
+		ctx,
+		valAddr,
+	)
```
Comment (toteki) [inline x/oracle/keeper/keeper_test.go:165]:
```suggestion
	app.OracleKeeper.DeleteAggregateExchangeRatePrevote(ctx, valAddr)
```

File: x/oracle/keeper/keeper_test.go [hunk 8]
```diff
@@ -0,0 +1,255 @@
...
+
+	_, err = app.OracleKeeper.GetAggregateExchangeRatePrevote(
+		ctx,
+		valAddr,
+	)
```
Comment (toteki) [inline x/oracle/keeper/keeper_test.go:170]:
```suggestion
	_, err = app.OracleKeeper.GetAggregateExchangeRatePrevote(ctx, valAddr)
```

File: x/oracle/keeper/keeper_test.go [hunk 9]
```diff
@@ -0,0 +1,255 @@
...
+
+	app.OracleKeeper.DeleteAggregateExchangeRateVote(
+		ctx,
+		valAddr,
+	)
+
+	_, err = app.OracleKeeper.GetAggregateExchangeRateVote(
+		ctx,
+		valAddr,
+	)
```
Comment (toteki) [inline x/oracle/keeper/keeper_test.go:207]:
```suggestion
	_, err = app.OracleKeeper.GetAggregateExchangeRateVote(ctx, valAddr)
```

File: x/oracle/keeper/keeper_test.go [hunk 10]
```diff
@@ -0,0 +1,255 @@
...
+	_, err := app.OracleKeeper.GetAggregateExchangeRateVote(
+		ctx,
+		valAddr,
+	)
+	s.Require().NoError(err)
+
+	app.OracleKeeper.DeleteAggregateExchangeRateVote(
+		ctx,
+		valAddr,
+	)
```
Comment (toteki) [inline x/oracle/keeper/keeper_test.go:202]:
```suggestion
	app.OracleKeeper.DeleteAggregateExchangeRateVote(ctx, valAddr)
```

File: x/oracle/keeper/keeper_test.go [hunk 11]
```diff
@@ -0,0 +1,255 @@
...
+	app.OracleKeeper.SetAggregateExchangeRateVote(
+		ctx,
+		valAddr,
+		vote,
+	)
+
+	_, err := app.OracleKeeper.GetAggregateExchangeRateVote(
+		ctx,
+		valAddr,
+	)
```
Comment (toteki) [inline x/oracle/keeper/keeper_test.go:196]:
```suggestion
	_, err := app.OracleKeeper.GetAggregateExchangeRateVote(ctx, valAddr)
```

File: x/oracle/keeper/keeper_test.go [hunk 12]
```diff
@@ -0,0 +1,255 @@
...
+
+	vote := types.AggregateExchangeRateVote{
+		ExchangeRateTuples: tuples,
+		Voter:              addr.String(),
+	}
+	app.OracleKeeper.SetAggregateExchangeRateVote(
+		ctx,
+		valAddr,
+		vote,
+	)
```
Comment (toteki) [inline x/oracle/keeper/keeper_test.go:191]:
```suggestion
	app.OracleKeeper.SetAggregateExchangeRateVote(ctx, valAddr, vote)
```

File: x/oracle/keeper/keeper_test.go [hunk 13]
```diff
@@ -0,0 +1,255 @@
...
+	app.OracleKeeper.SetExchangeRate(ctx, "umee", sdk.OneDec())
+	_, err := app.OracleKeeper.GetExchangeRate(ctx, "uumee")
+	s.Require().NoError(err)
+}
+
+func (s *IntegrationTestSuite) Test_DeleteExchangeRate() {
+	app, ctx := s.app, s.ctx
+
+	app.OracleKeeper.SetExchangeRate(ctx, "uumee", sdk.OneDec())
+	app.OracleKeeper.DeleteExchangeRate(ctx, "umee")
```
Comment (toteki) [inline x/oracle/keeper/keeper_test.go:250]: Might add a test afterward that confirms deletion occurred

Comment (toteki) [inline x/oracle/keeper/keeper_test.go:250]:
```suggestion
	app.OracleKeeper.DeleteExchangeRate(ctx, "umee")
	_, err := app.OracleKeeper.GetExchangeRate(ctx, "uumee")
	s.Require().NoError(err)
```

File: x/oracle/keeper/keeper_test.go [hunk 14]
```diff
@@ -0,0 +1,255 @@
...
+	_, err := app.OracleKeeper.GetExchangeRate(ctx, "uumee")
+	s.Require().Error(err)
+}
+
+func (s *IntegrationTestSuite) Test_GetExchangeRate_Valid() {
+	app, ctx := s.app, s.ctx
+
+	app.OracleKeeper.SetExchangeRate(ctx, "umee", sdk.OneDec())
+	_, err := app.OracleKeeper.GetExchangeRate(ctx, "uumee")
+	s.Require().NoError(err)
```
Comment (toteki) [inline x/oracle/keeper/keeper_test.go:217]:
```suggestion
	rate, err := app.OracleKeeper.GetExchangeRate(ctx, "uumee")
	s.Require().NoError(err)
	s.Require().Equal(sdk.OneDec(),rate)
```

File: x/oracle/keeper/keeper_test.go [hunk 15]
```diff
@@ -0,0 +1,255 @@
...
+	_, err = app.OracleKeeper.GetAggregateExchangeRateVote(
+		ctx,
+		valAddr,
+	)
+	s.Require().Error(err)
+}
+
+func (s *IntegrationTestSuite) Test_SetExchangeRateWithEvent() {
+	app, ctx := s.app, s.ctx
+	app.OracleKeeper.SetExchangeRateWithEvent(ctx, "umee", sdk.OneDec())
```
Comment (toteki) [inline x/oracle/keeper/keeper_test.go:187]:
```suggestion
	app.OracleKeeper.SetExchangeRateWithEvent(ctx, "umee", sdk.OneDec())
	
	rate, err := app.OracleKeeper.GetExchangeRate(ctx, "uumee")
	s.Require().NoError(err)
	s.Require().Equal(sdk.OneDec(),rate)
```

**Comment by toteki** [comment]
Besides the failing test (might be unrelated, hard to tell with panic message obscured) this should be good

**Comment by adamewozniak** [comment]
> Besides the failing test (might be unrelated, hard to tell with panic message obscured) this should be good

Looks like I forgot to add the oracle to `SetOrderInitGenesis`, should be fixed now

**Review by toteki — APPROVED** [review]

**Pull request merged**